Added release notes for 4.3.0 (#13923)
docker/docker.github.io: desktop/windows/release-notes/index.md

Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/projects/1){: target="_blank" rel="noopener" class="_"} to see what's coming next.

## Docker Desktop 4.3.0
2021-12-02

> Download Docker Desktop
>
> [For Windows](https://desktop.docker.com/win/main/amd64/Docker%20Desktop%20Installer.exe?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-win-amd64){: .button .primary-btn }

### Upgrades

- [Docker Engine v20.10.11](https://docs.docker.com/engine/release-notes/#201011)
- [containerd v1.4.12](https://github.com/containerd/containerd/releases/tag/v1.4.12)
- [Buildx 0.7.1](https://github.com/docker/buildx/releases/tag/v0.7.1)
- [Compose v2.2.1](https://github.com/docker/compose/releases/tag/v2.2.1)
- [Go 1.17.3](https://golang.org/doc/go1.17)
- [Kubernetes 1.22.4](https://github.com/kubernetes/kubernetes/releases/tag/v1.22.4)
- [hub-tool 0.4.4](https://github.com/docker/hub-tool/releases/tag/v0.4.4)

### Bug fixes and minor changes

> - Improved: Docker Desktop on Apple Silicon no longer requires Rosetta 2.

Reviewer: I don't think we should call out the Apple silicon Rosetta 2 entry in the Windows release notes

- Fixed an issue using the "Save As" action in the Volumes UI to save files from a volume. Fixes [docker/for-win#12407](https://github.com/docker/for-win/issues/12407).

> - Fixed problem when starting the Desktop UI in a home directory that contains characters used in regular expressions. Fixes [docker/for-win#12374](https://github.com/docker/for-win/issues/12374).

Reviewer: ```suggestion
- Fixed an issue that caused Docker Desktop to fail during startup if the home directory path contains a character used in regular expressions. Fixes [docker/for-win#12374](https://github.com/docker/for-win/issues/12374).
```

> - The following internal DNS names are considered deprecated and will be removed from a future release: `docker-for-desktop`, `docker-desktop`, `docker.for.mac.host.internal`, `docker.for.mac.localhost`, `docker.for.mac.gateway.internal`. Please use `host.docker.internal`, `vm.docker.internal` and `gateway.docker.internal`.

Reviewer: ```suggestion
- Deprecated: The following internal DNS names are deprecated and will be removed from a future release: `docker-for-desktop`, `docker-desktop`, `docker.for.mac.host.internal`, `docker.for.mac.localhost`, `docker.for.mac.gateway.internal`. You must now use `host.docker.internal`, `vm.docker.internal`, and `gateway.docker.internal`.
```

> - Removed custom RBAC rules. Fixes [docker/for-mac/#4774](https://github.com/docker/for-mac/issues/4774). When we first added k8s inside Docker Desktop, we added a rule to promote all service accounts to be cluster admin. It helped people who install helm to start easily and to forget security. We have now removed it.

Reviewer: How about "Removed: Custom RBAC rules have been removed from Docker Desktop as it gives `cluster-admin` privileges to all Service Accounts."

Not sure if we should explain the whole background here. @mat007 @StefanScherer WDYT?

Reviewer 2: Yes, yours is shorter and just as clear. I think I pasted the whole git commit message there because I was in a hurry when I made the PR. Sorry about that 😬

- Added a self-diagnose warning if the host lacks Internet connectivity.

> - Docker Desktop now uses cgroupv2. If you need to run `systemd` in a container then:

Reviewer: @trungutt this looks incomplete. Wondering if the following items (L48 and 49) are part of this issue?
